### PR TITLE
3860 by Inlead: Change contact form to use Site mail settings.

### PR DESCRIPTION
**Pull request by bozeak**
#### Link to issue

https://platform.dandigbib.org/issues/3860

#### Description

Changes the mail headers in order site's email settings are used.

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

### CODE REVIEW

File: modules/ding_contact/ding_contact.module [hunk 1]
```diff
@@ -64,3 +66,21 @@ function ding_contact_form_contact_site_form_alter(&$form, &$form_state) {
...
+/**
+ * Implements hook_mail_alter().
+ */
+function ding_contact_mail_alter(&$message) {
+  if('contact_page_mail' === $message['id']){
+    $subject = $message['subject'];
+    $default_from = variable_get('site_mail', ini_get('sendmail_from'));
+    $from_value =  t('"!subject via !site_name" <!site_mail>', array(
+      '!subject' => $subject,
+      '!site_name' => variable_get('site_name', 'Drupal'),
```
Comment (guddo) [inline modules/ding_contact/ding_contact.module]: Lets use "DDB CMS" instead of "Drupal".

File: modules/ding_contact/ding_contact.module [hunk 2]
```diff
@@ -64,3 +66,21 @@ function ding_contact_form_contact_site_form_alter(&$form, &$form_state) {
 function ding_contact_contact_form_stop_redirect($form, &$form_state) {
   unset($form_state['redirect']);
 }
+
+/**
+ * Implements hook_mail_alter().
+ */
+function ding_contact_mail_alter(&$message) {
+  if('contact_page_mail' == $message['id']){
```
Comment (NicolaeGrigoriu) [inline modules/ding_contact/ding_contact.module]: Add space after if and {
```suggestion
  if ('contact_page_mail' == $message['id']) {
```

File: modules/ding_contact/ding_contact.module [hunk 3]
```diff
@@ -47,6 +47,8 @@ function ding_contact_flooded_form($form, $form_state, $message) {
  * Why ? - to make sure no cached page is delivereed
  */
 function ding_contact_form_contact_site_form_alter(&$form, &$form_state) {
+  $message['from'] = variable_get('site_mail');
+  $message['headers']['Sender'] = variable_get('site_name', 'DDB CMS');
```
Comment (kasperg) [inline modules/ding_contact/ding_contact.module]: This refers to an undefined `$message` variable. That does not seem right?

Comment (bozeak) [reply modules/ding_contact/ding_contact.module]: @kasperg Thank you noticed. Probably just forgot to remove.

**Pull request merged**